src/emulator/modules/SceAppUtil/SceAppUtil.cpp

```diff
@@ -149,40 +150,37 @@ EXPORT(int, sceAppUtilResetCookieWebBrowser) {
     return UNIMPLEMENTED();
 }
 
-EXPORT(int, sceAppUtilSaveDataDataRemove, emu::SceAppUtilSaveDataFileSlot *slot, emu::SceAppUtilSaveDataRemoveItem *files, unsigned int fileNum, SceAppUtilSaveDataMountPoint *mountPoint) {
+const char *construct_savedata0_path(const char *data, const char *ext = "") {
+    return device::construct_normalized_path(VitaIoDevice::savedata0, data, ext);
+}
+
+const char *construct_slotparam_path(const unsigned int data) {
+    return construct_savedata0_path(std::string("SlotParam_" + std::to_string(data)).c_str(), "bin");
```

Comment:
Calling `c_str` on a string that's about to go out of scope is undefined behaviour, and I **think** that's what's happening here.

Reply:
That is actually resolved with [this](https://github.com/totlmstr/Vita3K/blob/io-paths/src/emulator/util/src/util.cpp#L161) function mentioned in `device::construct_normalized_path`.

Reply:
Nice. Is the result being freed later though?

Reply:
I ended up remodifying the function because it was being freed. Now, it is a `vector<char>` (which is recommended).